app/models/question.rb

```diff
@@ -83,9 +83,14 @@ class Question < ApplicationRecord
                      uniqueness: { scope: :section_id,
                                    message: UNIQUENESS_MESSAGE }
 
+  # =============
+  # = Callbacks =
+  # =============
 
+  # TODO: condition.remove_list needs to be serialized (from Array) before we can check
+  # for related conditions, so this can't be replaced by :destroy on the association
   before_destroy :check_remove_conditions
-  
```

Comment:
Because we've used a serialized-array field for the list of questions a condition removes, (rather than a join table), we'll need to create an instance of the conditions which may refer to this question (any condition in the template) and iteratively remove this question from each:
```

  def check_remove_conditions
    id = self.id.to_s
    self.template.questions.each do |q|
      q.conditions.each do |cond|
        cond.remove_data.delete(id)
        if cond.remove_data.empty?
          cond.destroy if cond.remove_data.empty?
        else
          cond.save
        end
      end
    end
  end
```

I think we may want to refactor this to an `after_destroy` to ensure the delete has actually worked,
But I'm again not sure how to re-create this functionality without a callback

Reply:
the placement of this code feels wrong ... an object manipulating it's peers before it's destroyed :/ ... maybe better if it lived in a service that the controller interacts with.

anyway, I think just leave as a callback for now

Reply:
I think the logic is correct but having to do this seems wrong. Seems to me that the "correct" thing to do here would be to use a join table. I stayed with the serialisation as closer to the original Killian implementation but it seems weird. Not sure it would be a huge task to rewrite it. Do now? Do later? Happy to go for it if consensus is do it now or leave till rails5 is done....

Reply:
agreed @raycarrick-ed. I think we should hold off on it until after the Rails 5 migration is complete